apps/genrsa.c

```diff
@@ -7,8 +7,11 @@
  * https://www.openssl.org/source/license.html
  */
 
+/* We need to use the deprecated RSA low level calls */
+#define OPENSSL_SUPPRESS_DEPRECATED
+
 #include <openssl/opensslconf.h>
-#ifdef OPENSSL_NO_RSA
+#if defined(OPENSSL_NO_RSA) || defined(OPENSSL_NO_DEPRECATED_3_0)
```

Comment:
Is this even necessary?

Reply:
I get build errors otherwise.  I'm not sure why, the file shouldn't be being built but seems to be.

Reply:
Okie, I'll have a look at that

Reply:
Perhaps if you remove them from the big `SOURCE[openssl]` a few lines up? :wink:

Reply:
I just noticed that you have forgotten that for DSA too (I failed to notice when reviewing, sorry)

Reply:
Sorry, I meant the big `$OPENSSLSRC` assignment

Reply:
Oops :flushed:

That would do it.

Reply:
Fix(es) pushed.